### PR TITLE
MTC Warning message fix

### PR DESCRIPTION
**Pull request by TomLonergan03**
## Description
Changed warning message system to be able to handle multiple errors

### CODE REVIEW

File: src/propulsion/controller.cpp [hunk 1]
```diff
@@ -438,44 +438,40 @@ void Controller::processSdoMessage(utils::io::can::Frame &message)
...
+      if (warning_message & 0x80) {
+        log_.ERR("MOTOR", "Controller %d warning: BMS proposed power");
+      }
+      if (warning_message & 0x100) {
+        log_.ERR("MOTOR", "Controller %d warning: Capacitor temperature exceeded");
+      }
+      if (warning_message & 0x200) { log_.ERR("MOTOR", "Controller %d warning: I2T protection"); }
+      if (warning_message & 0x400) {
+        log_.ERR("MOTOR", "Controller %d warning: Field weakening active ");
+      }
```
Comment (TomLonergan03) [inline src/propulsion/controller.cpp:472]: This formatting makes me sad but the auto formatting keeps returning it to this

File: src/propulsion/controller.cpp [hunk 2]
```diff
@@ -438,44 +438,40 @@ void Controller::processSdoMessage(utils::io::can::Frame &message)
...
-        case 0x200:
-          log_.ERR("MOTOR", "Controller %d warning: I2T protection");
-          break;
-        case 0x400:
-          log_.ERR("MOTOR", "Controller %d warning: Field weakening active ");
-          break;
-        default:
-          log_.ERR("MOTOR", "Controller %d warning: Warning code %d", node_id_, warning_message);
+      uint16_t warning_message
+        = stoi(std::to_string(message.data[5]) + std::to_string(message.data[4])); // Merge error bytes into one string
```
Comment (TomLonergan03) [inline src/propulsion/controller.cpp:442]: This feels kinda hacky for concatenating bytes but seems to be the best way to do it

File: src/propulsion/controller.cpp [hunk 3]
```diff
@@ -438,44 +438,40 @@ void Controller::processSdoMessage(utils::io::can::Frame &message)
...
+      }
+      if (warning_message & 0x04) {
+        log_.ERR("MOTOR", "Controller %d warning: DC link under voltage");
+      }
+      if (warning_message & 0x08) {
+        log_.ERR("MOTOR", "Controller %d warning: DC link over voltage");
+      }
+      if (warning_message & 0x10) { log_.ERR("MOTOR", "Controller %d warning: DC over current"); }
+      if (warning_message & 0x20) {
+        log_.ERR("MOTOR", "Controller %d warning: Stall protection active");
```
Comment (MiltFra) [inline src/propulsion/controller.cpp:458]: You're missing an argument for `%d` here.

**Review by MiltFra — CHANGES_REQUESTED** [review]

**Review by MiltFra — APPROVED** [review]

File: src/propulsion/controller.cpp [hunk 4]
```diff
@@ -438,44 +438,44 @@ void Controller::processSdoMessage(utils::io::can::Frame &message)
...
-        case 0x200:
-          log_.ERR("MOTOR", "Controller %d warning: I2T protection");
-          break;
-        case 0x400:
-          log_.ERR("MOTOR", "Controller %d warning: Field weakening active ");
-          break;
-        default:
-          log_.ERR("MOTOR", "Controller %d warning: Warning code %d", node_id_, warning_message);
+      // Merge error bytes into one variable
+      uint16_t warning_message = (message.data[5] << 8) + message.data[4];
```
Comment (adenhaus) [inline src/propulsion/controller.cpp:442]: const?

**Review by kshxtij — APPROVED** [review]
LGTM

**Pull request merged**